src/App/App.test.js

```diff
@@ -179,17 +181,119 @@ describe('App', () => {
   });
 
   it('should be able to go to the favorites page', async () => {
+    fetchTodaysColor.mockResolvedValueOnce({color: 'mock color'});
     const { getByText } = render(
       <MemoryRouter>
         <App />
       </MemoryRouter>
     );
 
-    const favorites = await waitFor(() => getByText('view my gallery'));
-    
-    fireEvent.click(favorites);
+    fireEvent.click(await waitFor(() => getByText('view my gallery')));
     
     expect(getByText('no favorites yet!')).toBeInTheDocument();
   });
 
+  it('should call localStorage setItem when favorite or unfavorite is clicked', async () => {
+    fetchTodaysColor.mockResolvedValueOnce({color: 'mock color'});
+    getArt.mockResolvedValueOnce(mockArt);
+    const { getByText, getByTestId } = render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+    
+    fireEvent.click(await waitFor(() => getByText('view today\'s gallery')));
+    fireEvent.click(await waitFor(() => getByTestId('fave2')));
+
+    expect(localStorage.setItem).toBeCalledTimes(1);
+  });
+
+  it('should be able to go to a piece\'s page from favorites', async () => {
+    fetchTodaysColor.mockResolvedValueOnce({color: 'mock color'});
+    getArt.mockResolvedValueOnce(mockArt);
+    const { getByText, getByTestId } = render(
```

Comment:
Wondering if this test does too much? I don't know how I could have broken it up though, and it does make sure all the discrete parts work together.